source/common/http/utility.cc

```diff
@@ -73,17 +73,11 @@ void Utility::appendXff(HeaderMap& headers, const Network::Address::Instance& re
     return;
   }
 
-  HeaderString& header = headers.insertForwardedFor().value();
-  const std::string& address_as_string = remote_address.ip()->addressAsString();
-  HeaderMapImpl::appendToHeader(header, address_as_string.c_str());
+  headers.appendForwardedFor(remote_address.ip()->addressAsString(), ",");
 }
 
 void Utility::appendVia(HeaderMap& headers, const std::string& via) {
-  HeaderString& header = headers.insertVia().value();
-  if (!header.empty()) {
-    header.append(", ", 2);
-  }
-  header.append(via.c_str(), via.size());
+  headers.appendVia(via, ", ");
```

Comment:
I vaguely recall discussing this with @alyssawilk at some point, but why do we append via with a comma and space, but forwarded for only a comma? Should we be consistent?

Reply:
No clue why we append via with whitespce - I suspect it's legacy from before we had the append function.
I'd be happy to have it consistent, though given the mess the last time we tried to remove whitespace I'd suggest 
1) validating Envoy via parsing doesn't expect whitespace and
2) runtime guarding in case someone else has upstreams which expect it

Interestingly another oh so mystery server I happen to code on has a utility function for append and append-with-whitespace and Via is appended with whitespace - I wonder if that too is legacy or if there's some common issue specific to VIA that some standard server screws up whitespace or something.  I don't know the history internally or externally though :-/

Reply:
Added a TODO to investigate this. In theory we could eliminate the delimiter specification if it can the typical `,` append